chef/cookbooks/keystone/libraries/helpers.rb

```diff
@@ -3,8 +3,18 @@ def self.service_URL(node, host, port)
     "#{node[:keystone][:api][:protocol]}://#{host}:#{port}"
   end
 
-  def self.versioned_service_URL(node, host, port)
-    service_URL(node, host, port) + '/' + node[:keystone][:api][:version] + '/'
+  def self.versioned_service_URL(node, host, port, version=nil, current_node=nil, cookbook_name=nil)
```

Comment:
I still believe we should simply not pass `node` anymore here if we want this to be usable outside of this cookbook, just the bits we need (protocol, version, etc.). There's no need to try to keep the API compatible as it's only used by the keystone cookbook right now, so you could fix it right away.